docs: Add Readme for Release (#84)
CUTR-at-USF/MUSER: BUILD.md

> # Building the project

Reviewer: Could you please update the README as well to remove the redundant sections there and point to this BUILD.md document?

We use [Gradle build flavors](http://developer.android.com/tools/building/configuring-gradle.html#workBuildVariants) to enable a number of different build variants of MUSER.

For now, We have one  Gradle "platform" flavor dimensions:

> * **Google** = Normal Google Play release

Reviewer: It doesn't look like there is actually a `google` platform flavor in MUSER the way it's defined in OBA, so I wouldn't say that here. Instead, Shuttle defines the variants along a single dimension `all` in `build.gradle` and then filters out some of the combined variants programatically in `build.gradle` for those that don't make sense (e.g., `devRelease`).

For clarity, please list all the build variants that are possible in the app, and then specify which we are using for MUSER. You can see the final possible combined variants using the "Build variants" window in Android Studio:

![image](https://user-images.githubusercontent.com/928045/96181383-e79ad500-0f01-11eb-9056-af60a87ba94d.png)

Looks like:
* `amazonFreeRelease`
* `amazonPaidRelease`
* `devDebug`
* `freeRelease`
* `paidRelease`

IIRC the only ones that matter for MUSER are `devDebug` and `freeRelease`.

Author: Thanks for putting this together. I guess I'll mention these variants and document what we are currently using in the BUILD.md file.

> ...and one Gradle "brand" flavor dimensions:

Reviewer: See above - there is no `brand` flavor dimension. I would just copy/paste the text from my above comment describing how the variants are generated, list all possible combined ones, and then just explain which matter to MUSER.

Author: Also, should I use the word "Shuttle" or should I say "MUSER"?

Reviewer: I would say that this app is derived from Shuttle, which has these variant combinations. But, in the case of MUSER, these are the variants that we're using.

* **free** = Original MUSER brand
* **debug** = Debug variant of the MUSER

We have one build variant with one release and one debug type resulting in 2 build variants. 

> The below instructions assume you're going to be building for the **Google** platform flavor and original **debug** brand by default (e.g., `devDebug`).

Reviewer: ```suggestion
The below instructions assume you're going to be building the `debug` variant for testing on Google devices (i.e.,, `devDebug`).
```

### Prerequisites for both Android Studio and Gradle

1. Clone this repository
1. Install [Java Development Kit (JDK)](http://www.oracle.com/technetwork/java/javase/downloads/index.html)

### Building in Android Studio

1. Download, install, and run the latest version of [Android Studio](http://developer.android.com/sdk/installing/studio.html).
1. At the welcome screen select `Import Project`, browse to the location of this repository and select it then select Ok.
1. Open the Android SDK Manager (Tools->Android->SDK Manager) and add a checkmark for the necessary API level (see `compileSdkVersion` in [`MUSER/build.gradle`](MUSER/build.gradle)) then select OK.
1. Connect a [debugging enabled](https://developer.android.com/tools/device.html) Android device to your computer or setup an Android Virtual Device (Tools->Andorid->AVD Manager).
1. Open the "Build Variants" window (it appears as a vertical button on left side of workspace by default) & choose **devDebug** to select the Google Play version.
1. Click the green play button (or Alt+Shift+F10) to build and run the project!

### Building from the command line using Gradle

> 1. Set the `JAVA_HOME` environmental variables to point to your JDK folder (e.g. `C:\Program Files\Java\jdk1.6.0_27`)

Reviewer: We want people using JDK 7 or higher, so I'd prefer to reference 7 instead of 6 in the example:

```suggestion
1. Set the `JAVA_HOME` environmental variables to point to your JDK folder (e.g. `C:\Program Files\Java\jdk1.7.0_27`)
```

1. To build and push the app to the device, run `gradlew installdevDebug` from the command line at the root of the project.
1. To start the app, run `adb shell am start -n com.simplecity.amp_library.ui.screen.main.MainActivity` (alternately, you can manually start the app)

### Release builds

To build a release build, you need to create a `gradle.properties` file that points to a `secure.properties` file, and a `secure.properties` file that points to your keystore and alias.

The `gradle.properties` file is located in the `MUSER` directory and has the contents:
```
secure.properties=<full_path_to_secure_properties_file>
```

The `secure.properties` file (in the location specified in gradle.properties) has the contents:
```
key.store=<full_path_to_keystore_file>
key.alias=<key_alias_name>
key.storepassword=<your_keystore_password>
key.keypassword=<your_key_password>
```

Note that the paths in these files always use the Unix path separator `/`, even on Windows. If you use the Windows path separator `\` you will get the error `No value has been specified for property 'signingConfig.keyAlias'.`

Then, to build all flavors run:

> `gradlew assembleRelease`

Reviewer: Since we never use any other release variants in the case of MUSER, I'd explain this and then just use `gradlew assembleFreeRelease` here.
